### DISCUSSION TITLE
p270, p291 인증관련 질문입니다

### DISCUSSION THREAD

**Discussion by Onthesly** [fsoftwareengineer/todo-application#30] [질문과 답변] [answered]
질문을 작성하실 때에는 아래의 내용을 포함 해 주시기 바랍니다.
오류나 에러가 나는 페이지
1) p.270 1. "그림 4-15 위조된 토큰 사용" 처럼 body가 표시되지 않습니다. 
          2. 예제 4-15 위조된 토근 로그 밑에 긴 에러로그가 뜹니다.
그 외에 인증에 성공하였을때는 문제 없이 잘됩니다.
2) p.290  코드작성후 실행시 콘솔창에 Uncaught (in promise) SyntaxError: Unexpected end of JSON input라는 에러가 뜨면서
           리다이렉트(/'login')가 되지 않습니다.
           postman으로 GET했을때 body부분이 비어있는데 이때문에 파싱을 못해서라고 짐작하고 있습니다...

오류/에러 로그
p.270 관련 
![image](https://user-images.githubusercontent.com/87744184/146150249-ed511424-a7a2-48a6-9f13-7e9105d63b4a.png)
p.290 관련
![image](https://user-images.githubusercontent.com/87744184/146151814-9958734d-fe6b-4d66-bf54-cad0b0cf2b3c.png)
postman GET요청시 또는 잘못된 토큰이용시
![image](https://user-images.githubusercontent.com/87744184/146152335-67f61a01-bc70-4052-8b59-f1e869c4679c.png)
![image](https://user-images.githubusercontent.com/87744184/146152790-09b7e781-466c-4eae-8869-ef11bacc6631.png)

작성 중인 코드
p.290 까지 깃허브코드 참조하여 똑같이 작성하였습니다.

오류/에러를 해결하기 위해 현재까지 시도 해 본 방법들
1) TokenProvider.java에 SECRET_KEY -> SECRET_KEY.getBytes(Charset.forName("UTF-8")) 로 변경해봄
2) GetMapping에 try,catch문을 넣어 catch문에 ResponseEntity.badRequest().body(response); 리턴하도록 해봄
3) 하루종일 구글링해가며 찾아봤는데 영어가 부족해서인지 해결법을 모르겠습니다...ㅠ

**Comment by kimym56** [comment, marked as answer]
build.gradle에서 spring boot 버전이 어떻게 되시나요?

**Reply by Onthesly** [reply]
plugins {
	id 'org.springframework.boot' version '2.6.1'
	id 'io.spring.dependency-management' version '1.0.11.RELEASE'
	id 'java'
} 입니다

**Reply by kimym56** [reply]
2.5.7로 변경하시고 Refresh Gradle Project 해보시겠어요? 왜그러는지는 모르겠는데 그렇게 해결했습니다..

**Reply by Onthesly** [reply]
와 해결됐어요!! 정말 감사합니다!!! 버전차이때문에 이게 안될줄이야ㅠㅠ
진도 나갈수있게 해주셔서 정말 감사합니다!!!!!!

**Reply by Onthesly** [reply]
https://github.com/spring-projects/spring-boot/issues/28953
영어를 잘 몰라서 확실하지는 않지만 관련된 이슈 같습니다.
링크 첨부합니다.

**Reply by mw7895la** [reply]
혹시 Gradle 버전은 6점대 이신가요?

**Reply by Onthesly** [reply]
저는 7.3.1 이네요

**Reply by jojojojocho** [reply]
감사합니다 ㅠㅠ 덕분에 해결했습니다. 스프링 2.6.3버전에서 같은 문제가 발생했습니다. 다들 참고하세요!!

**Reply by indiesloth** [reply]
저는 버전을 내렸는데도 계속 문제가 발생하네요..ㅠㅠ 이걸로 며칠째인지

**Reply by silver0-stack** [reply]
저도 버전도 2.6.8로 내려도 login 페이지로 redirect가 안됩니다 
@indiesloth  해결하셨나요?

**Reply by indiesloth** [reply]
@silver0-stack 버전에 문제가 있었던 것이 아니라 코드에 문제가 있었습니다. 답장이 늦어서 미안해요.

**Comment by fsoftwareengineer** [comment]
https://github.com/fsoftwareengineer/todo-application/discussions/41
에서도 비슷한 문제가 있었던 것 같아 참고용으로 남깁니다.
